index.html

```diff
@@ -418,6 +419,163 @@ <h2>Transferable MediaStreamTrack</h2>
       <p>The underlying source is supposed to be kept alive between the transfer and transfer-receiving steps, or as long as the data holder is alive.
       In a sense, between these steps, the data holder is attached to the underlying source as if it was a track.</p>
     </div>
+    <h3>MediaStreamTrack frame stats</h3>
+    <div>
+      <p>
+        On any track that supports {{MediaTrackSettings/frameRate}} as a
+        setting, frame counters allow the application to tell what the frame
+        rate is, which may be lower than the target
+        {{MediaTrackSettings/frameRate}}. For example, if the track is sourced
+        from a camera then the production of frames could be slowed down if it's
+        dark or frames could be dropped if the system is CPU starved. This could
+        impact the total number of frames handled by the source and impact how
+        many frames are delivered, discarded or lost.
+      </p>
+    </div>
+    <div>
+      <pre class="idl"
+>partial interface MediaStreamTrack {
+  Promise&lt;MediaTrackFrameStats&gt; getFrameStats();
+};
+</pre>
+    </div>
+    <div>
+      <p>If a {{MediaStreamTrack}} supports {{MediaTrackSettings/frameRate}} as
+      a setting, the user agent is required to count each frame from its
+      source as follows:</p>
+      <ul>
+        <li>
+          <p>A frame is considered
+          <dfn data-lt="delivered frames">delivered</dfn> if it either was
+          delivered to a sink or would have been delivered to a sink, if one was
+          connected. This is a subset of [= total frames =] and it is
+          incremented at the same time as [= total frames =].</p>
+        </li>
+        <li>
+          <p>A frame is considered
+          <dfn data-lt="discarded frames">discarded</dfn> if it was discarded in
+          order to achieve the target {{MediaTrackSettings/frameRate}}.
+          This is a subset of [= total frames =] and it is incremented at the
+          same time as [= total frames =].</p>
+        </li>
+        <li>
+          <p>The <dfn data-lt="total frames">total</dfn> number of frames that
+          have been processed by this source, meaning it is known whether the
+          frame was considered delivered, discarded or dropped for any other
+          reason. The number of dropped frames can be calculated by subtracting
+          [= delivered frames =] and [= discarded frames =] from
+          [= total frames =].</p>
+          <div class="note">
```

Comment:
We haven't defined what dropped means. It seems to me that dropped is a catch-all category that is anything that's not categorized as delivered or discarded.

Reply:
But it is meant to be a catch-all category for anything that is not categorized as delivered or discarded, so it seems the definition fits the bill.

If we want to figure out why a frame was lost we could add more counters later, but different user agents may drop for different reasons. The interesting thing about total frames is that it tells you the camera frame rate, which is not measurable without this metric.

Reply:
perhaps:

_The number of dropped frames for various unknown reasons can be calculated by subtracting_

Reply:
Done